tutorial/common.py

```diff
@@ -0,0 +1,132 @@
+
+    def on_help_button_clicked(self, _):
+        self.output.value = f"""<p style="color:RoyalBlue;">Hint: {self.hint}</p>"""
+
+    def clear(self):
+        self.answer.value = None
+        self.clear_output()  # if the value above was already None, the output will not be cleared
+
+    def clear_output(self, _=None):
+        """Clears the output when the answer is changed."""
+        self.output.value = ''
+    
+    def correct_answer_given(self):
+        return self.answer.label == self.correct_answer
+    
+    def answer_is_given(self):
+        return self.answer.value is not None
+
+class Quiz(ipw.VBox):
+    def __init__(self, questions: list=None) -> None:
+        """A Quiz widget.
+            
+            questions: A list of Question widgets.
+        """
+        self.verify_button = ipw.Button(description='Verify', button_style='success')
+        self.verify_button.on_click(self.verify)
+
+        self.clear_button = ipw.Button(description='Clear', button_style='danger')
+        self.clear_button.on_click(self.clear)
+
+        self.output = ipw.HTML()
+
+        self.aux = [ipw.HBox([self.verify_button, self.clear_button]), self.output]
+
+        super().__init__()
+
+        self.nquestions = 0
+        self.questions = []
+        for question in questions:
+            self.add_question(question)
+    
+    def add_question(self, question: Question):
+        """Adds a question to the quiz."""
+        question.question.value = f"""<strong>Q{self.nquestions+1}:</strong> """ + question.question.value
+        self.questions.append(question)
+        self.children = self.questions + self.aux
+        self.nquestions += 1
+    
+    def questions_not_answered(self):
+        """Returns a list of question numbers that are not answered."""
+        to_return = []
+        for i, question in enumerate(self.questions):
+            if not question.answer_is_given():
+                question.request_answer_if_necessary()
+                to_return.append(i+1)
+        return to_return
+
+    def erroneous_responses(self):
+        """Returns a list of question numbers that are answered incorrectly."""
+        to_return = []
+        for i, question in enumerate(self.questions):
+            if not question.verify():
+                to_return.append(i+1)
+        return to_return
+
+    def verify(self, _=None):
+        """Verifies the answers and displays the results."""
+
+        # Dealing with unanswered questions.
+        not_answered = self.questions_not_answered()
+        if not_answered:
+            self.output.value = f"""<p style="color:Tomato;">Not answerred questions: {", ".join(map(str, not_answered))}!</p>"""
+            return
+
+        # Dealing with erroneous responses.
+        errors = self.erroneous_responses()
+        if errors:
+            self.output.value = f"""<p style="color:Tomato;">Incorrect answers: {", ".join(map(str, errors))}!</p>"""
```

Comment:
I would only change the output string when there are incorrect answers. Right now, it says `Incorrect answers: <question number>`. Even a small change like
```html
<p style="color:Tomato;">Your answers to the following questions are <strong>incorrect</strong>: Q1, Q2</p>
```
will be clear enough.

Reply:
thanks, fixed in 86f74a0